url/src/lib.rs

```diff
@@ -2734,30 +2733,48 @@ fn path_to_file_url_segments_windows(
                 let share = share.to_str().ok_or(())?;
                 serialization.extend(percent_encode(share.as_bytes(), PATH_SEGMENT));
             }
-            _ => return Err(()),
+            Prefix::DeviceNS(_device_name) => {
+                // Paths starting with "\\.\" (Local Device Paths) are intentionally not supported.
+                return Err(());
+            }
+            _ => {
+                host_end = to_u32(serialization.len()).unwrap();
+                host_internal = HostInternal::None;
+                serialization.push('/');
+            }
         },
-
-        _ => return Err(()),
+        Some(Component::Normal(_)) | Some(Component::ParentDir) => return Err(()),
+        _ => {
+            host_end = to_u32(serialization.len()).unwrap();
+            host_internal = HostInternal::None;
+            // serialization.push('/');
+            // dbg!(&host_end, &serialization);
+            // return Err(());
```

Comment:
Is the commented text left on purpose?
Could you make sure this branch is covered by a unit test?

Reply:
Yes, this is still a draft PR - I'm just getting an idea of what we expect the new behavior to be and then once I'm done I'll mark as ready for review.